pkg/services/app/add_test.go

```diff
@@ -100,7 +106,43 @@ var _ = Describe("Add", func() {
 		Expect(deployKey).To(Equal([]byte("deploy key")))
 	})
 
+	Describe("checks for existing deploy key before creating secret", func() {
+		It("looks up deploy key and skips creating secret if found", func() {
+			defaultParams.SourceType = string(app.SourceTypeGit)
+
+			gitProviders.DeployKeyExistsStub = func(s1, s2 string) (bool, error) {
+				return true, nil
+			}
+
+			err := appSrv.Add(defaultParams)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(gitProviders.UploadDeployKeyCallCount()).To(Equal(0))
```

Comment:
it would be nice to check the flux calls since that's what actually breaks the code, so adding `Expect(fluxClient.CreateSecretGitCallCount()).To(Equal(0|1))` where is due

Reply:
Makes sense.